airflow/metrics/otel_logger.py

```diff
@@ -235,6 +301,57 @@ def del_counter(self, name: str, attributes: Attributes = None) -> None:
         if key in self.map.keys():
             del self.map[key]
 
+    def set_value(self, name: str, value: float, delta: bool, tags: Attributes):
+        """
+        Overrides the last reading for a Gauge with a new value.
+
+        :param name: The name of the gauge to record.
+        :param value: The new reading to record.
+        :param delta: If True, value is added to the previous reading, else it overrides.
+        :param tags: Gauge attributes which were used to generate a unique key to store the counter.
+        :returns: None
+        """
+        key: str = _generate_key_name(name, tags)
+        old_value = self.poke_gauge(name, tags)
+        # If delta is true, add the new value to the last reading otherwise overwrite it.
+        self.map[key] = Observation(value + (delta * old_value), tags)
```

Comment:
oh wow because `True == 1` ? And the type checkers don't say anything ? 😱

Reply:
Yup, python will happily treat True as 1; None and False are 0.  :)   

Another handy use for that is if you want to see how many Trues you have, you can use sum, like "How many names contain the letter A?" could be `sum(['A' in name for name in names])`  You'll end up with a list[bool] containing both True and False values, and sum() will add all the 1's and 0's accordingly.

Reply:
While neat, I find this a little obtuse/esoteric. I'd just go with a more straightforward if-statement or ternary operator so that it is more readable for future folks. Or at the very least a comment above this line explaining it.

Reply:
Alright, I've changed it to an if block to make it more obvious.